app/server/lib/HostedStorageManager.ts

```diff
@@ -748,14 +779,18 @@ export class HostedStorageManager implements IDocStorageManager {
       await this._inventory.uploadAndAdd(docId, async () => {
         const prevSnapshotId = this._latestVersions.get(docId) || null;
         const newSnapshotId = await this._ext.upload(docId, tmpPath as string, metadata);
+        snapshotProgress.lastWindowDoneAt = Date.now();
```

Comment:
This whole method is async, and you are mutating the the `snapshotProgress` that might be used somewhere else (also in async). 
But I don't know if that is bad or not here, might be fine.

Reply:
Yes, this is intentional, the fields of snapshotProgress gather different events across time and aren't from a single moment. The flow is pretty complex, and I rejected a bunch of measurements that would have not have had a coherent meaning. The things I chose to mention work I *think*.